### PR TITLE
[Keyboard] Add theone h75 keyboard

### PR DESCRIPTION
**Pull request by Persama**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dunk2k** [comment]
Missing a `readme.md` file 🙈

**Comment by Persama** [comment]
Okay, thank you. I have added the `readme.md` file.

File: keyboards/theone/h75/ansi_tkl/ansi_tkl.c [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+{
+    m_side_led_show();
+
+    return true;
+}
+
+void housekeeping_task_user(void)
+{
+    long_press_key();
+}
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/ansi_tkl.c:666]:
```suggestion
void housekeeping_task_kb(void)
{
    long_press_key();
    housekeeping_task_user();
}
```

File: keyboards/theone/h75/ansi_tkl/ansi_tkl.c [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+
+            device_reset_init();
+        }
+    } else {
+        reset_press_time = 0;
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record)
+{
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/ansi_tkl.c:510]:
```suggestion
bool process_record_kb(uint16_t keycode, keyrecord_t *record)
{
    if (!process_record_user(keycode, record)) {
        return false;
    }
```

File: keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	KC_LSFT,				KC_Z,   	KC_X,   	KC_C,  		KC_V,   	KC_B,   	KC_N,   	KC_M,   	KC_COMM,	KC_DOT,		KC_SLSH,				KC_RSFT,	KC_UP,
+	KC_LCTL,	KC_LALT,	KC_LGUI,										KC_SPC, 							KC_RGUI,	MO(1),   	KC_RCTL,	KC_LEFT,	KC_DOWN,    KC_RGHT),
+// layer Mac Fn
+[1] = LAYOUT(
+	_______, 	KC_F1,  	KC_F2,  	KC_F3, 		KC_F4,  	KC_F5,  	KC_F6,  	KC_F7,  	KC_F8,  	KC_F9, 		KC_F10, 	KC_F11, 	KC_F12, 				KC_INS,
+	_______, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	RGB_TOG, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	SYS_WIN,  	SYS_MAC,   	DEV_RESET,	_______, 	_______,	_______,
+	_______,	RGB_VAI,  	RGB_VAD,  	RGB_MOD,   	RGB_HUI, 	RGB_SPI, 	RGB_SPD,	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	KC_END,
+	_______,				SIDE_VAI,  	SIDE_VAD,  	SIDE_MOD,  	SIDE_HUI,	SIDE_SPI, 	SIDE_SPD, 	_______,   	_______,	_______,	_______,				_______,	_______,
+	_______,	_______,	_______,										_______, 							_______,	MO(1),   	_______,	_______,	_______,    _______),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c:22]:
```suggestion
	_______,	_______,	_______,										_______, 							_______,	_______,   	_______,	_______,	_______,    _______),
```

File: keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+// layer win
+[2] = LAYOUT(
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c:24]: This layer does not seem to be accessible..

File: keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	KC_LSFT,				KC_Z,   	KC_X,   	KC_C,  		KC_V,   	KC_B,   	KC_N,   	KC_M,   	KC_COMM,	KC_DOT,		KC_SLSH,				KC_RSFT,	KC_UP,	
+	KC_LCTL,	KC_LGUI,	KC_LALT,										KC_SPC, 							KC_RALT,	MO(3),   	KC_RCTL,	KC_LEFT,	KC_DOWN,    KC_RGHT),
+// layer win Fn
+[3] = LAYOUT(
+	_______, 	KC_BRID,   	KC_BRIU,    _______,  	_______,   	_______,   	_______,   	KC_MPRV,   	KC_MPLY,   	KC_MNXT,  	KC_MUTE, 	KC_VOLD, 	KC_VOLU,				KC_INS,	
+	_______, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	RGB_TOG, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	SYS_WIN,  	SYS_MAC,   	DEV_RESET,	_______, 	_______,	_______,
+	_______,	RGB_VAI,  	RGB_VAD,  	RGB_MOD,   	RGB_HUI, 	RGB_SPI, 	RGB_SPD,	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	KC_END,
+	_______,				SIDE_VAI,  	SIDE_VAD,  	SIDE_MOD,  	SIDE_HUI,	SIDE_SPI, 	SIDE_SPD, 	_______,   	_______,	_______,	_______,				_______,	_______,
+	_______,	_______,	_______,										_______, 							_______,	MO(3),   	_______,	_______,	_______,    _______),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c:38]:
```suggestion
	_______,	_______,	_______,										_______, 							_______,	_______,   	_______,	_______,	_______,    _______),
```

File: keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	_______,				_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,	_______,	_______,				_______,	_______,
+	_______,	_______,	_______,										_______, 							_______,	MO(1),   	_______,	_______,	_______,    _______),
+// layer 7
+[7] = LAYOUT(
+	_______, 	_______,  	_______,  	_______, 	_______,  	_______,  	_______,  	_______,  	_______,  	_______, 	_______, 	_______, 	_______, 				_______,
+	_______, 	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______, 	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______,	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	_______,
+	_______,				_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,	_______,	_______,				_______,	_______,
+	_______,	_______,	_______,										_______, 							_______,	MO(1),   	_______,	_______,	_______,    _______),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c:70]:
```suggestion
```

File: keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04
+} };
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c:108]: This does not belong at the keymap level and needs to be migrated to `info.json` file.
```suggestion
```

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+	KC_LSFT,				KC_Z,   	KC_X,   	KC_C,  		KC_V,   	KC_B,   	KC_N,   	KC_M,   	KC_COMM,	KC_DOT,		KC_SLSH,				KC_RSFT,	KC_UP,
+	KC_LCTL,	KC_LALT,	KC_LGUI,										KC_SPC, 							KC_RGUI,	MO(1),   	KC_RCTL,	KC_LEFT,	KC_DOWN,    KC_RGHT),
+// layer Mac Fn
+[1] = LAYOUT(
+	_______, 	KC_F1,  	KC_F2,  	KC_F3, 		KC_F4,  	KC_F5,  	KC_F6,  	KC_F7,  	KC_F8,  	KC_F9, 		KC_F10, 	KC_F11, 	KC_F12, 				KC_INS,
+	_______, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	SYS_WIN,  	SYS_MAC,   	DEV_RESET,	_______, 	_______,	_______,
+	_______,	_______,  	_______,  	_______,   	_______, 	_______, 	_______,	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	KC_END,
+	_______,				_______,  	_______,  	_______,  	_______,	_______, 	_______, 	MO(4),   	RGB_SPD,	RGB_SPI,	_______,				_______,	RGB_VAI,
+	_______,	_______,	_______,										_______, 							_______,	MO(1),   	_______,	RGB_MOD,	RGB_VAD,    RGB_HUI),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:22]:
```suggestion
	_______,	_______,	_______,										_______, 							_______,	_______,   	_______,	RGB_MOD,	RGB_VAD,    RGB_HUI),
```

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+// layer win
+[2] = LAYOUT(
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:24]: This layer does not seem accessible..

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+	KC_LSFT,				KC_Z,   	KC_X,   	KC_C,  		KC_V,   	KC_B,   	KC_N,   	KC_M,   	KC_COMM,	KC_DOT,		KC_SLSH,				KC_RSFT,	KC_UP,	
+	KC_LCTL,	KC_LGUI,	KC_LALT,										KC_SPC, 							KC_RALT,	MO(3),   	KC_RCTL,	KC_LEFT,	KC_DOWN,    KC_RGHT),
+// layer win Fn
+[3] = LAYOUT(
+	_______, 	KC_BRID,   	KC_BRIU,    _______,  	_______,   	_______,   	_______,   	KC_MPRV,   	KC_MPLY,   	KC_MNXT,  	KC_MUTE, 	KC_VOLD, 	KC_VOLU,				KC_INS,	
+	_______, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______, 	_______,   	_______,    _______,  	_______,   	_______,   	_______,   	_______,   	_______,   	SYS_WIN,  	SYS_MAC,   	DEV_RESET,	_______, 	_______,	_______,
+	_______,	_______,  	_______,  	_______,   	_______, 	_______, 	_______,	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	KC_END,
+	_______,				_______,  	_______,  	_______,  	_______,	_______, 	_______, 	MO(4),   	RGB_SPD,	RGB_SPI,	_______,				_______,	RGB_VAI,
+	_______,	_______,	_______,										_______, 							_______,	MO(3),   	_______,	RGB_MOD,	RGB_VAD,    RGB_HUI),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:38]:
```suggestion
	_______,	_______,	_______,										_______, 							_______,	_______,   	_______,	RGB_MOD,	RGB_VAD,    RGB_HUI),
```

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+// layer 4
+[4] = LAYOUT(
+	_______, 	_______,  	_______,  	_______, 	_______,  	_______,  	_______,  	_______,  	_______,  	_______, 	_______, 	_______, 	_______, 				_______,
+	_______, 	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______, 	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______,	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	_______,
+	_______,				_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	SIDE_SPD,	SIDE_SPI,	_______,				_______,	SIDE_VAI,
+	_______,	_______,	_______,										_______, 							_______,	MO(4),   	_______,	SIDE_MOD,	SIDE_VAD,   SIDE_HUI),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:46]:
```suggestion
	_______,	_______,	_______,										_______, 							_______,	_______,   	_______,	SIDE_MOD,	SIDE_VAD,   SIDE_HUI),
```

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+	_______,				_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,	_______,	_______,				_______,	_______,
+	_______,	_______,	_______,										_______, 							_______,	MO(1),   	_______,	_______,	_______,    _______),
+// layer 7
+[7] = LAYOUT(
+	_______, 	_______,  	_______,  	_______, 	_______,  	_______,  	_______,  	_______,  	_______,  	_______, 	_______, 	_______, 	_______, 				_______,
+	_______, 	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______, 	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,   	_______,	_______, 	_______,	_______,
+	_______,	_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,   	_______,  	_______,	_______, 	 			_______,	_______,
+	_______,				_______,   	_______,   	_______,  	_______,   	_______,   	_______,   	_______,   	_______,	_______,	_______,				_______,	_______,
+	_______,	_______,	_______,										_______, 							_______,	MO(1),   	_______,	_______,	_______,    _______),
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:70]:
```suggestion
```

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
+    0x04
+} };
+
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:109]:
```suggestion
```

File: keyboards/theone/h75/ansi_tkl/rules.mk [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+AUDIO_ENABLE = no            # Audio output
+KEY_LOCK_ENABLE = yes
+ENCODER_ENABLE = no
+
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes
+
+WS2812_DRIVER = bitbang
+
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/rules.mk:16]:
```suggestion
# This file intentionally left blank
```

File: keyboards/theone/h75/ansi_tkl/info.json [hunk 15]
```diff
@@ -0,0 +1,116 @@
+{
+    "keyboard_name": "NuPhy Krypton75",
+    "manufacturer": "github.com/sowbug",
+    "usb":{
+        "vid": "0x19F5",
+        "pid": "0x3245",
+        "device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/theone/h75/ansi_tkl/info.json:8]:
```suggestion
    },
    "features": {
        "bootmagic": true,
        "mousekey": true,
        "extrakey": true,
        "console": false,
        "command": false,
        "nkro": true,
        "key_lock": true,
        "rgb_matrix": true
    },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c (hunk 12 above)
Comment (Persama) [reply keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:70]: I would like to reserve these layers for VIA

Comment (lendunistus) [reply keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:70]: you don't need to make filler layers for VIA, it's handled automatically

**Comment by waffle87** [comment]
Please **do not** mark suggestions as resolved if you have not done anything to address them, it makes the reviewing process extremely difficult. I have marked a number of them as unresolved.

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c (hunk 12 above)
Comment (Persama) [reply keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:70]: Okay, thank you

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c (hunk 13 above)
Comment (Persama) [reply keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:85]: The light definition has been deleted, can it still work properly? It should be placed in another file,

**Comment by Persama** [comment]
Okay, thank you for your guidance @waffle87

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c (hunk 13 above)
Comment (waffle87) [reply keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:85]: From my other comment
> This does not belong at the keymap level and needs to be migrated to `info.json` file.

File: keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c (hunk 4 above)
Comment (Persama) [reply keyboards/theone/h75/ansi_tkl/keymaps/default/keymap.c:24]: Switch access to this layer through special button processing

File: keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c (hunk 9 above)
Comment (Persama) [reply keyboards/theone/h75/ansi_tkl/keymaps/via/keymap.c:24]: Switch access to this layer through special button processing

File: keyboards/theone/h75/ansi_tkl/config.h
Comment (dunk2k) [inline keyboards/theone/h75/ansi_tkl/config.h:1]: can settings, in this  file, that have a data-driven equivalent be moved to `info.json`?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/theone/h75/ansi_tkl/config.h
Comment (Persama) [reply keyboards/theone/h75/ansi_tkl/config.h:1]: Okay, I will re optimize and standardize the code.

**Pull request closed without merging**